io/net/ethtool_test.py

```diff
@@ -131,14 +135,31 @@ def test_ethtool(self):
         for state, status in zip(["down", "up"], ["no", "yes"]):
             if not self.interface_state_change(self.iface, state, status):
                 self.fail("interface %s failed" % state)
-            cmd = "ethtool %s %s %s" % (self.args, self.iface, self.elapse)
-            ret = process.run(cmd, shell=True, verbose=True,
-                              ignore_status=True)
-            if ret.exit_status != 0:
-                if "Operation not supported" in ret.stderr_text:
-                    self.log.warn("%s failed" % self.args)
-                else:
-                    self.fail("failed")
+            if self.args=="-L" :
+                if self.other!='' and self.combined!='' :
+                    cmd1 = "ethtool %s %s tx %s rx %s other %s combined %s" % (self.args, self.iface, self.tx, self.rx, self.other, self.combined)
+                else :
+                    cmd1 = "ethtool %s %s tx %s rx %s" % (self.args, self.iface, self.tx, self.rx)
+                ret1 = process.run(cmd1, shell=True, verbose=True,
+                                  ignore_status=True)
+                if ret1.exit_status != 0:
+                    if "Invalid argument" in ret1.stderr_text:
+                        self.fail("Invalid argument")
+                cmd2 = "ethtool %s %s %s" % ("-l", self.iface, self.elapse)
+                ret2 = process.run(cmd2, shell=True, verbose=True,
+                                  ignore_status=True)
+                stdout = ret2.stdout_text
+                if (stdout.find("self.tx") !=-1) and (stdout.find("self.rx") !=-1):
+                    self.fail("channel set failed")
+            else:
+                cmd = "ethtool %s %s %s" % (self.args, self.iface, self.elapse)
+                ret = process.run(cmd, shell=True, verbose=True,
+                                  ignore_status=True)
+                if ret.exit_status != 0:
+                    if "Operation not supported" in ret.stderr_text:
+                        self.log.warn("%s failed" % self.args)
+                    else:
+                        self.fail("failed")
```

Comment:
@TasmiyaNalatwad print the failing arg too. it will help for quick debug

Reply:
Printing arg value along with failed statement in case of failure.